pkgs/servers/akkoma/pleroma-fe/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, stdenv
+, fetchFromGitea, fetchYarnDeps
+, fixup_yarn_lock, yarn, nodejs
+, jpegoptim, oxipng, nodePackages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pleroma-fe";
+  version = "2022-09-10";
+
+  src = fetchFromGitea {
+    domain = "akkoma.dev";
+    owner = "AkkomaGang";
+    repo = "pleroma-fe";
+    rev = "d7499a1f91fb5e4ec947445c566f8ae41cec6c72";
+    hash = "sha256-I+d+oTOTNaMPkk9vdFHhm/g8EtPTZnsWv8yPUa0uNHQ=";
+  };
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
+    hash = "sha256-UkSdVRqFp6bj7zz4ZRMD47tVpuS7x7F1MiEei+87kyg=";
+  };
+
+  nativeBuildInputs = [
+    fixup_yarn_lock
+    yarn
+    nodejs
+    jpegoptim
+    oxipng
+    nodePackages.svgo
+  ];
+
+  # Build scripts assume to be used within a Git repository checkout
+  patchPhase = ''
+    sed -E -i \
+      -e '/^let commitHash =/,/;$/clet commitHash = "${builtins.substring 0 7 src.rev}";' \
+      build/webpack.prod.conf.js
+  '';
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export HOME="$PWD/tmp"
+    mkdir -p $HOME
+
+    yarn config --offline set yarn-offline-mirror ${lib.escapeShellArg offlineCache}
+    fixup_yarn_lock yarn.lock
+
+    yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+    patchShebangs node_modules/
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    export NODE_ENV="production"
+    export NODE_OPTIONS="--openssl-legacy-provider"
+    yarn run build --offline
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # (Losslessly) optimise compression of image artifacts
+    find dist -type f -name '*.jpg' -execdir ${jpegoptim}/bin/jpegoptim -w$NIX_BUILD_CORES {} \;
+    find dist -type f -name '*.png' -execdir ${oxipng}/bin/oxipng -o max -t $NIX_BUILD_CORES {} \;
+    find dist -type f -name '*.svg' -execdir ${nodePackages.svgo}/bin/svgo {} \;
```

Comment:
How long is this taking? If we are building for a long time then we should skip this.

Reply:
This part takes a total of 1.48 s on my i7-11850H with `NIX_BUILD_CORES=4`. Is that acceptable?

Reply:
thats not very long. I was just afraid that this might take 30 minutes because there are to many files or so.